src/theme/TOC/index.js

```diff
@@ -2,25 +2,21 @@
 
 import React from "react";
 import clsx from "clsx";
-import useTOCHighlight from "@theme/hooks/useTOCHighlight";
+import { useTOCHighlight } from "@docusaurus/theme-common/internal";
 import styles from "./styles.module.css";
 const LINK_CLASS_NAME = "table-of-contents__link";
 const ACTIVE_LINK_CLASS_NAME = "table-of-contents__link--active";
 const TOP_OFFSET = 100;
 /* eslint-disable jsx-a11y/control-has-associated-label */
 
 function Headings({ toc, isChild }) {
-  if (!toc.length) {
-    return null;
-  }
```

Comment:
Should we restore this check with `?` added? `if (!toc?.length) {`

This would keep the old behavior of the component to return null, and then we could remove `?` from the `toc.map` below.